compiler/src/dotty/tools/dotc/semanticdb/ExtractSemanticDB.scala

```diff
@@ -130,61 +125,52 @@ class ExtractSemanticDB extends Phase:
 
       tree match
         case tree: PackageDef =>
-          if !excludeDef(tree.pid.symbol)
-          && tree.pid.span.hasLength then
-            tree.pid match
-            case tree: Select =>
-              registerDefinition(tree.symbol, selectSpan(tree), Set.empty, tree.source)
-              traverse(tree.qualifier)
-            case tree => registerDefinition(tree.symbol, tree.span, Set.empty, tree.source)
           tree.stats.foreach(traverse)
+          if !excludeDef(tree.pid.symbol) && tree.pid.span.hasLength then
```

Comment:
Sry, there's some indent change..., pls see the diff with `?w=1`...

Reply:
Modified the `traverse` function so that it traverses **child nodes first**, and then tries to register the Symbol of the current node.

This is required because in `TypeOps`, we construct the symbol table for `LambdaType` and `RecOrRefined`, and when we resolve the symbols for `ParamRef` (in `LambdaType`) and `refinement` (in `RefinedType` or `RecType`) by looking up that symbol table.

For example, if we have

```scala
type Person = Record {
  val name: String
  val age: Int
}
```

To convert the refinement type for `Person` (`RefinedType(RefinedType(..., name, ...), age, ...)`) to `StructuralType(...)`, we have to know the symbol for `name` and `age`, which we cannot obtain purely from that type information.
To know the symbols for `name` and `age`, we
- traverse the child nodes of `type Person` to register the symbol of `name` and `age` to the symbol table
- and then convert the `RefinedType` of `Person` with resolving the symbols for `name` and `age` by looking up the symbol table.